### PR TITLE
use nludataimporter to avoid domain errors in nlu testing

### PR DESCRIPTION
**Pull request by erohmensing**
**Proposed changes**:
- Use an `NluDataImporter` on a `RasaFileImporter` directly instead of loading up a full TrainingDataImporter for NLU testing. 
- Previously, we called [this function](https://github.com/RasaHQ/rasa/blob/main/rasa/cli/test.py#L162) without a config param, so it was automatically using a `RasaFileImporter` no matter what. By making this change, we are still using a RasaFileImporter, but only the NLU part, and avoiding also [adding a bunch of responses stuff, e2e, etc](https://github.com/RasaHQ/rasa/blob/db3cb7d52ab30e39633a0b0fdfd85f64b66178df/rasa/shared/importers/importer.py#L162) which AFAIK shouldn't be needed here (i could be wrong though) 
- close https://github.com/RasaHQ/rasa/issues/8848

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by erohmensing** [comment]
@ancalita I'm not sure about testing this one. Most of the cli tests use [these fixtures](https://github.com/RasaHQ/rasa/blob/7625fcaec9a5d1f47dbcc093a405e9369e7261fc/tests/cli/conftest.py#L66-L99), and it doesnt seem super scalable to make a new one for each project structure I'm trying to test. Any ideas?

**Comment by ancalita** [comment]
> it doesnt seem super scalable to make a new one for each project structure I'm trying to test.

@erohmensing How many project structures were you considering? I think it's fine to test if the default structure still works (I think that's tested [here](https://github.com/RasaHQ/rasa/blob/7625fcaec9a5d1f47dbcc093a405e9369e7261fc/tests/cli/test_rasa_test.py#L103) already?) and another test for a structure that has the domain in a different subdirectory as you mentioned in the issue?

Unrelated - what caused so many Poetry errors in the CI tests? I've never seen this before.

File: rasa/cli/test.py [hunk 1]
```diff
@@ -158,9 +158,7 @@ async def run_nlu_test_async(
     )
 
     data_path = rasa.cli.utils.get_validated_path(data_path, "nlu", DEFAULT_DATA_PATH)
-    test_data_importer = TrainingDataImporter.load_from_dict(
-        training_data_paths=[data_path], domain_path=DEFAULT_DOMAIN_PATH,
-    )
+    test_data_importer = NluDataImporter(RasaFileImporter(training_data_paths=[data_path]))
```
[the remark this answers is not in the document]
Comment (erohmensing) [reply rasa/cli/test.py:161]: I tried this (without changing the ofiginal importers) to no avail 😕

**Comment by erohmensing** [comment]
> Unrelated - what caused so many Poetry errors in the CI tests? I've never seen this before.

It's github being down, I think , we can re-run the tests tomorrow

File: rasa/cli/test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ancalita) [reply rasa/cli/test.py:161]: Oh I meant in the change you made, shouldn't you provide the `training_type` argument too as part of `RasaFileImporter`'s constructor?

**Comment by wochinge** [comment]
FYI that we release 2.8 today / tomorrow. So please point this to `main` or `2.8.x` then

**Comment by ancalita** [comment]
@erohmensing All good to close this now?

**Comment by wochinge** [comment]
Since Ella is no longer here, good to close 😞

**Pull request closed without merging**